server/src/server.ts

```diff
@@ -425,6 +407,49 @@ connection.onCompletion((pos: CompletionParams): CompletionItem[] => {
     return completionProvider.getAvailableCompletionItems(pos.textDocument.uri, doc.offsetAt(pos.position));
 });
 
+connection.onDocumentFormatting((_params) => {
+    const doc = documents.get(_params.textDocument.uri);
+    // We should directly send an empty list if this feature is disabled by the settings.
+    return buildDocumentEditList(doc, SETTINGS.documentFormat);
+});
+
+/**
+ * Build and send the list of Text Edit to apply to `document`
+ * accordingly to the settings defined in `documentFormatSettings`.
+ *
+ * @param document the text document to format
+ * @param documentFormatSettings the document format settings to apply
+ */
+export function buildDocumentEditList(document: TextDocument, documentFormatSettings?: IDocumentFormatSettings) {
```

Comment:
I first tried to put this code in another `feature` file but it failed because of some obscure cross dependency issues during the texts. No problem when this code is here. :shrug: